src/video-player/components/runtime.tsx

```diff
@@ -57,55 +57,57 @@ export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, set
-              const currentCue = captionsTrackRef.current?.activeCues?.length && captionsTrackRef.current?.activeCues.length > 0 && captionsTrackRef.current?.activeCues[0];
-              log("cue change", { videoUrl: authoredState.videoUrl, currentCue: (currentCue as VTTCue).text });
-            });
+    if (playerRef.current) {
+      const player: videojs.Player = videojs(playerRef.current,
+        {
+          controls: true,
+          fluid: !(authoredState.fixedAspectRatio || authoredState.fixedHeight),
+          // This is a new property not supported by the current types
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          crossOrigin: "anonymous"
+        }, () => {
+          const url = authoredState.videoUrl ? authoredState.videoUrl : "";
+          player.src(url);
+
+          if (authoredState.captionUrl && player.textTracks().length === 0) {
+            player.addRemoteTextTrack({
+              kind: "captions",
+              language: "en",
+              label: "English",
+              src: authoredState.captionUrl,
+              default: captionsOnByDefault
+            }, false);
+            // Store a ref to the captions track s we can check visibility later. There is no easy toggle event for captions show/hide
+            captionsTrackRef.current = player.textTracks()[0];
+            if (captionsTrackRef.current) {
+              captionsTrackRef.current.addEventListener("cuechange", () => {
+                const currentCue = captionsTrackRef.current?.activeCues?.length && captionsTrackRef.current?.activeCues.length > 0 && captionsTrackRef.current?.activeCues[0];
+                log("cue change", { videoUrl: authoredState.videoUrl, currentCue: (currentCue as VTTCue).text });
+              });
+            }
           }
-        }
-      });
+        });
 
-    if (authoredState.fixedAspectRatio) {
-      const aspectRatio = getAspectRatio(authoredState.fixedAspectRatio);
-      if (aspectRatio.length > 0) {
-        player.aspectRatio(aspectRatio);
+      if (authoredState.fixedAspectRatio) {
+        const aspectRatio = getAspectRatio(authoredState.fixedAspectRatio);
+        if (aspectRatio.length > 0) {
+          player.aspectRatio(aspectRatio);
+        }
       }
-    }
-    if (authoredState.fixedHeight) {
-      player.height(authoredState.fixedHeight);
-    }
-    player.ready(() => {
-      if (viewedTimestampRef.current) {
-        player.currentTime(viewedTimestampRef.current);
+      if (authoredState.fixedHeight) {
+        player.height(authoredState.fixedHeight);
       }
-    });
+      player.ready(() => {
+        if (viewedTimestampRef.current) {
+          player.currentTime(viewedTimestampRef.current);
+        }
+      });
 
-    return () => {
-      player.dispose();
-    };
+      return () => {
+        player.dispose();
+      };
+    }
   }, [authoredState]);
```

Comment:
Need to add a dependency to `playerRef.current` in the array

Reply:
> Need to add a dependency to `playerRef.current` in the array

Maybe -- refs are generally used specifically to store values that shouldn't trigger re-renders on change. I take your point to be that the effect should run again when `playerRef.current` changes (which makes sense) but before reflexively adding `playerRef.current` to the dependencies I'd ask whether the value should really be in a ref. If it should trigger a re-render on change then maybe it should be state, for instance.

Reply:
I think what @dougmartin is pointing out is that in this code:
```typescript
  useEffect(() => {
    if (playerRef.current) {
      const player: videojs.Player = videojs(playerRef.current,
```
If `playerRef.current` were `undefined` at the time of the first render, then if it subsequently became defined we would want the effect to run again. The fact that things work currently suggests that either `playerRef.current` is always defined when the effect runs or that if it ever changes that change also correlates with other changes that trigger a re-render. I suspect that it's always defined on first render and the only time it becomes `undefined` is on unmount, in which case we don't really care whether the effect runs or not. The easiest thing to do in the short run is just to leave it alone since it seems to be working, but if we ever need to handle the case of transitioning from `undefined` to defined then we'd have to do something to make sure that the effect runs in that case.

Reply:
OK, thanks. I've left the dependency array as is in the latest commit since it doesn't seem necessary to include `playerRef`. 

This may not need to be stated, but for the record, if I remove `if (playerRef.current) {`, the line immediately after it causes the error `Argument of type 'HTMLVideoElement | null' is not assignable to parameter of type 'string | Element'. Type 'null' is not assignable to type 'string | Element'.` to be thrown (where `Argument of type 'HTMLVideoElement | null'` refers to `playerRef.current`).

Let me know if you still think `playerRef` should be included, though, @dougmartin @lublagg.